cli/js/compiler.ts

```diff
@@ -843,6 +847,46 @@ const ignoredDiagnostics = [
   7016,
 ];
 
+type Stats = Array<{ key: string; value: number }>;
+
+const stats: Stats = [];
+let statsStart = 0;
+
+function performanceStart(): void {
+  stats.length = 0;
+  // TODO(kitsonk) replace with performance.mark() when landed
+  statsStart = performance.now();
+  ts.performance.enable();
+}
+
+function performanceProgram(program: ts.Program): void {
+  stats.push({ key: "Files", value: program.getSourceFiles().length });
+  stats.push({ key: "Nodes", value: program.getNodeCount() });
+  stats.push({ key: "Identifiers", value: program.getIdentifierCount() });
+  stats.push({ key: "Symbols", value: program.getSymbolCount() });
+  stats.push({ key: "Types", value: program.getTypeCount() });
+  stats.push({ key: "Instantiations", value: program.getInstantiationCount() });
+  const programTime = ts.performance.getDuration("Program");
+  const bindTime = ts.performance.getDuration("Bind");
+  const checkTime = ts.performance.getDuration("Check");
+  const emitTime = ts.performance.getDuration("Emit");
+  stats.push({ key: "Parse time", value: programTime });
+  stats.push({ key: "Bind time", value: bindTime });
+  stats.push({ key: "Check time", value: checkTime });
+  stats.push({ key: "Emit time", value: emitTime });
```

Comment:
Use the same keys as used in TS API?
So `s/Emit time/Emit/` and `s/Check time/Check/` and so on.